### ISSUE TITLE
Prune location.ancestorOrigins entries

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#33671]
[`location.ancestorOrigins`](https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins) allows embedded pages to know the origin of all of the parent pages.

For example, if `a.com` iframes `b.com` which itself iframes `c.com`, then the innermost frame (`c.com`) can see: `[b.com, a.com]`

Previously, only the immediate parent was visible via the `Referer` header or `document.referrer` property.

Firefox has not yet implemented this, but is [considering](https://bugzilla.mozilla.org/show_bug.cgi?id=1085214#c26) doing so in a way that does not leak `a.com` to `c.com`:
- show the parent origin unless the referrer would be omitted based on `a.com`'s referrer policy
- only expose the number of other ancestor origins

In our example, Firefox would therefore hypothetically return `[b.com, "null"]` where [`"null"`](https://github.com/whatwg/html/issues/1918#issuecomment-254336641) does include the quotes.

**Comment by fmarier** [comment]
Test page: https://fmarier.com/referrer/frames.html

**Comment by ShivanKaul** [comment]
@pes10k suggested checking Chrome web data for prevalence.

**Comment by pes10k** [comment]
here is the chrome web data info: https://chromestatus.com/metrics/feature/timeline/popularity/823

TL;DR; it looks very uncommon, and so its either a very low risk, or Chrome stoped instrumenting it.

**Comment by tophf** [comment]
> or Chrome stoped instrumenting it.

Yes, it stopped in 2017, and can be enabled only explicitly by making a build with compile-time flags enable_blink_bindings_tracing, extended_tracing_enabled. You can see the new histogram by git-blaming the [source](https://crsrc.org/out/chromeos-Debug/gen/third_party/blink/renderer/bindings/core/v8/v8_location.cc;l=80;drc=250488700300de84991d19fbb8c6b7ddad9e4760).
